demos/sample_azure_iot/sample_azure_iot.c

```diff
@@ -0,0 +1,625 @@
+                                                       ( const uint8_t * ) sampleazureiotMESSAGE,
+                                                       sizeof( sampleazureiotMESSAGE ) - 1,
+                                                       &xPropertyBag );
+            configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+            LogInfo( ( "Attempt to receive publish message from IoTHub.\r\n" ) );
+            xResult = AzureIoTHubClient_ProcessLoop( &xAzureIoTHubClient,
+                                                     sampleazureiotPROCESS_LOOP_TIMEOUT_MS );
+            configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+            if( ulPublishCount % 2 == 0 )
+            {
+                /* Send reported property every other cycle */
+                xResult = AzureIoTHubClient_SendDeviceTwinReported( &xAzureIoTHubClient,
+                                                                    ( const uint8_t * ) sampleazureiotTWIN_PROPERTY,
+                                                                    sizeof( sampleazureiotTWIN_PROPERTY ) - 1,
+                                                                    NULL );
+                configASSERT( xResult == eAzureIoTHubClientSuccess );
+            }
+
+            /* Leave Connection Idle for some time. */
+            LogInfo( ( "Keeping Connection Idle...\r\n\r\n" ) );
+            vTaskDelay( sampleazureiotDELAY_BETWEEN_PUBLISHES_TICKS );
+        }
+
+        xResult = AzureIoTHubClient_UnsubscribeDeviceTwin( &xAzureIoTHubClient );
+        configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+        xResult = AzureIoTHubClient_UnsubscribeDirectMethod( &xAzureIoTHubClient );
+        configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+        xResult = AzureIoTHubClient_UnsubscribeCloudToDeviceMessage( &xAzureIoTHubClient );
+        configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+        /* Send an MQTT Disconnect packet over the already connected TLS over
+         * TCP connection. There is no corresponding response for the disconnect
+         * packet. After sending disconnect, client must close the network
+         * connection. */
+        xResult = AzureIoTHubClient_Disconnect( &xAzureIoTHubClient );
+        configASSERT( xResult == eAzureIoTHubClientSuccess );
+
+        /* Close the network connection.  */
+        TLS_Socket_Disconnect( &xNetworkContext );
+
+        /* Wait for some time between two iterations to ensure that we do not
+         * bombard the broker. */
+        LogInfo( ( "Demo completed successfully.\r\n" ) );
+        LogInfo( ( "Short delay before starting the next iteration.... \r\n\r\n" ) );
+        vTaskDelay( sampleazureiotDELAY_BETWEEN_DEMO_ITERATIONS_TICKS );
+    }
+}
+/*-----------------------------------------------------------*/
+
+#ifdef democonfigENABLE_DPS_SAMPLE
+
+    /**
+    * @brief Get IoTHub endpoint and device Id info, when Provisioning service is used.
```

Comment:
Nit - I think we should doc that this might take "a while" to run - like seconds, presumably configured by some #define macro?